internal/core/data/controller/messaging/subscriber.go

```diff
@@ -114,11 +114,11 @@ func unmarshalPayload(envelope types.MessageEnvelope, target interface{}) error
 }
 
 func validateEvent(messageTopic string, e dtos.Event) errors.EdgeX {
-	// Parse messageTopic by the pattern `edgex/events/<device-profile-name>/<device-name>/<source-name>`
+	// Parse messageTopic by the pattern `edgex/events/device/<device-service-name>/<device-profile-name>/<device-name>/<source-name>`
 	fields := strings.Split(messageTopic, "/")
 
 	// assumes a non-empty base topic with /profileName/deviceName/sourceName appended by publisher
-	if len(fields) < 4 {
+	if len(fields) < 5 {
```

Comment:
@lenny-intel Do you think we still need to check this ? I didn't see the change in #4348

Reply:
I believe we need this check, and update the comment in line 120
however, why doesn't it `if len(fields) < 6`?